synapse/cortex.py

```diff
@@ -1966,8 +1982,47 @@ async def syncLayerNodeEdits(self, iden, offs, wait=True):
         if layr is None:
             raise s_exc.NoSuchLayer(iden=iden)
 
-        async for item in layr.syncNodeEdits(offs, wait=wait):
-            yield item
+        async for offs, nodeedits, _ in layr.syncNodeEdits(offs, wait=wait):
+            yield (offs, nodeedits)
+
+    async def mergedLayerNodeEdits(self, offs=None):
+        '''
+        Yield (layr, offs, mesg) tuples for nodeedits in all layers.
+        '''
+        if offs is None:
+            offs = {}
+
+        async with await s_base.Base.anit() as base:
+
+            outq = asyncio.Queue(maxsize=1000)
+
+            async def layredits(_iden, _offs):
+                layr = self.getLayer(_iden)
+                if layr is None:
+                    raise s_exc.NoSuchLayer(iden=_iden)
+
+                async for loff, mesg, meta in layr.syncNodeEdits(_offs):
+                    await outq.put((_iden, loff, mesg, meta))
```

Comment:
May need to merge-sort the outputs based on offset.  If they end up in the queue out of offs order, the consumer can't track the offset and provide it back

Reply:
I considered that but I think it should be fine since the edits come out of the layer in order and the consumer should be tracking the offsets per layer to provide back.